Develop

yoshida-prog opened the pull request:
TodoアプリをVueで作成
Vue-Cli（vuex）を使用

疑問に思った箇所はコード内にコメントアウトしております。
ご確認のほどよろしくお願いいたします。

yoshida-prog committed:
Update index.js

yoshida-prog committed:
Update App.vue

Kenichiro-Kiwaki commented on src/App.vue:
```diff
@@ -1,11 +1,70 @@
+    },
+    // ラジオボタンに応じて全てのtodoの状態を確認してソート
+    checkState(todo){
+      if(this.radio === 'すべて'){
```
- `{}`の両外側
- ifの後ろの`()`の両外側
上記にはスペースをあけるようにしましょう。
他の場所でも同様です。
```suggestion
      if (this.radio === 'すべて') {
```
自動整形を使うといいでしょう。

Kenichiro-Kiwaki commented on src/store/index.js:
```diff
@@ -5,15 +5,29 @@ Vue.use(Vuex)
+      if(comment){
+        // mutations内ではgettersが使用できないため直接state.todosにpushした
+        // 以下の2つのmutationについても同じように直接state.todosを使っているが
+        // gettersを使う必要なないのでしょうか？？
```
gettersはstateを参照するために使用するので、
今回のようにstateの状態を変更する際には使う必要はありません。

mutationは同期的でなければいけませんので、
stateの状態を変更するおに非同期処理を行う場合には
actionsを経由させる必要があります。

https://vuex.vuejs.org/ja/guide/mutations.html

Kenichiro-Kiwaki commented on src/App.vue:
```diff
@@ -1,11 +1,70 @@
+          <th>コメント</th>
+          <th>状態</th>
+        </tr>
+        <tr v-for="(todo, index) in $store.state.todos" :key="index" v-show="checkState(todo)">
```
v-ifとv-forをいっしょに使うことは進められていません。
その理由から考えると、v-showとv-forも併用するのは避けたほうが良いでしょう。

https://jp.vuejs.org/v2/style-guide/#v-for-%E3%81%A8%E4%B8%80%E7%B7%92%E3%81%AB-v-if-%E3%82%92%E4%BD%BF%E3%81%86%E3%81%AE%E3%82%92%E9%81%BF%E3%81%91%E3%82%8B-%E5%BF%85%E9%A0%88

Kenichiro-Kiwaki reviewed (commented):
コメントいたしました！

yoshida-prog committed:
Update App.vue

v-forとv-showを同一要素で使用している箇所を修正

yoshida-prog committed:
Update index.js

atom-Beautifyを使用してコードを自動整形

yoshida-prog commented:
・コードの自動整形
・v-forとv-showを同一要素に使用していた箇所を修正

yoshida-prog committed:
Update App.vue

不要な要素がありましたので削除しました。

yoshida-prog commented:
不要な要素がありましたので削除しました。

Kenichiro-Kiwaki commented on src/App.vue:
```diff
@@ -1,32 +1,36 @@
+        <th>コメント</th>
+        <th>状態</th>
+      </tr>
+      <tr v-for="(todo, index) in $store.state.todos" :key="index">
```
こちら前回の指摘漏れになってしまい申し訳ありません。
予期せぬ変更が起こらないように、stateは直接参照せずにgetterを使用しましょう。

Kenichiro-Kiwaki commented on src/App.vue:
```diff
@@ -1,32 +1,36 @@
+        <th>状態</th>
+      </tr>
+      <tr v-for="(todo, index) in $store.state.todos" :key="index">
+        <template v-if="checkState(todo)">
```
この実装方法ですと、v-forで処理した要素すべてにv-ifで条件つけレンダリングしているので非効率的です。
例えばtodosをフィルタリングしてdoingTodoやdoneTodoなどの配列を用意し、それをv-forで処理するようにしましょう。
https://jp.vuejs.org/v2/style-guide/#v-for-%E3%81%A8%E4%B8%80%E7%B7%92%E3%81%AB-v-if-%E3%82%92%E4%BD%BF%E3%81%86%E3%81%AE%E3%82%92%E9%81%BF%E3%81%91%E3%82%8B-%E5%BF%85%E9%A0%88

Kenichiro-Kiwaki reviewed (commented):
コメントいたしました！

yoshida-prog committed:
Update App.vue

・todoリストを算出プロパティでフィルタリングをして表示するよう修正
・stateの呼び出しはゲッターを使用

yoshida-prog committed:
Update index.js

・stateにaddIDを追加（todoの追加及び削除の際に使用）
・deleteTodo関数内に変数redefinitionIDを追加（todoの削除後にリストのIDを再定義する際に使用）

yoshida-prog commented on src/store/index.js:
```diff
@@ -5,20 +5,32 @@ Vue.use(Vuex)
   state: {
-    todos: []
+    todos: [],
+    addID: 0
```
・stateにaddIDを追加（todoの追加及び削除の際に使用）

yoshida-prog commented on src/store/index.js:
```diff
@@ -5,20 +5,32 @@ Vue.use(Vuex)
+      for (let todo of state.todos) {
+        todo.id = redefinitionID
+        redefinitionID++
+      }
```
・deleteTodo関数内に変数redefinitionIDを追加（todoの削除後にリストのIDを再定義する際に使用）

yoshida-prog commented on src/App.vue:
```diff
@@ -32,38 +30,36 @@
-      }
+    }
+  },
+  computed: {
+    checkState() {
+      return this.$store.getters.getTodos.filter(todo => {
+        if (this.radio === 'すべて') {
+          return todo
+        } else if (this.radio === '作業中') {
+          return todo['State'] === '作業中'
+        } else if (this.radio === '完了') {
+          return todo['State'] === '完了'
+        }
```
算出プロパティでtodoリストのフィルタリングをして表示させた。
stateはゲッター呼び出した。

Kenichiro-Kiwaki replied:
可読性を上げるために、this.$store.getters.getTodos を変数に格納し、その後filterにかけるほうがより良いですね。

yoshida-prog committed:
Update App.vue
